private/buf/cmd/buf/command/curl/curl.go

```diff
@@ -420,18 +420,36 @@ func (f *flags) validate(isSecure bool) error {
 		return fmt.Errorf("grpc protocol cannot be used with plain-text URLs (http) unless --%s flag is set", http2PriorKnowledgeFlagName)
 	}
 
+	if f.Schema != "" && f.Reflect {
+		if f.flagSet.Changed(reflectFlagName) {
+			// explicitly enabled both
+			return fmt.Errorf("cannot specify both --%s and --%s", schemaFlagName, reflectFlagName)
+		}
+		// Reflect just has default value; unset it since we're going to use --schema instead
+		f.Reflect = false
+	}
+	if !f.Reflect && f.Schema == "" {
+		return fmt.Errorf("must specify --%s if --%s is false", schemaFlagName, reflectFlagName)
+	}
+	schemaIsStdin := strings.HasPrefix(f.Schema, "-")
```

Comment:
I moved this up so that it happens before the other flag checks. Mainly, there's the possibility this logic _changes_ the value of `f.Reflect` (line 429). So we need that settled before we use that value to check other flags.